aggregate_root/lib/aggregate_root.rb

```diff
@@ -37,10 +37,8 @@ def apply(*events)
 
   def load(stream_name, event_store: default_event_store)
     @loaded_from_stream_name = stream_name
-    events_enumerator(event_store, stream_name).with_index do |event, index|
-      apply(event)
-      @version = index
-    end
+    event_store.read.stream(stream_name).reduce {|_, ev| apply(ev) }
```

Comment:
@mpraglowski should  this still be in batches?

Reply:
Since https://github.com/RailsEventStore/rails_event_store/commit/6d0d204c46039775e86fcf72c142b6ee809fadcf it is already in batches.